business_rules/operators.py

```diff
@@ -801,13 +801,14 @@ def empty(self, other_value: dict):
     def empty_within_except_last_row(self, other_value: dict):
         target = self.replace_prefix(other_value.get("target"))
         comparator = other_value.get("comparator")
+        order_by_column: str = self.replace_prefix(other_value.get("ordering"))
         # group all targets by comparator
-        grouped_target = self.value.groupby(comparator)[target]
+        ordered_df = self.value.sort_values(by=[comparator, order_by_column])
+        grouped_target = ordered_df.groupby(comparator)[target]
         # validate all targets except the last one
         results = grouped_target.apply(lambda x: x[:-1]).apply(lambda x: x in ["", None])
-        # extract values with corresponding indexes from results
-        self.value[f"result_{uuid4()}"] = results.reset_index(level=0, drop=True)
-        return True in results.values
+        # return values with corresponding indexes from results
+        return pd.Series(results.reset_index(level=0, drop=True))
```

Comment:
We should add documentation for the added param https://github.com/cdisc-org/cdisc-rules-engine/blob/main/resources/schema/Operator.md#empty_within_except_last_row

Reply:
Done, see PR on rules-engine.